### PR TITLE
Fix test error in rooted_sbn_instance.hpp

### LINKED ISSUE #280: Tests fail on OSX/Clang
```
_build/rooted_sbn_instance.hpp:170: ERROR: CHECK_EQ( correct_parameters, inst.PrettyIndexedSBNParameters() ) is NOT correct!
  values: CHECK_EQ( [(00111, 0.25), (01111, 0.5), (00010, 0.25), (00100|01010|00010, 1), (00111|11000|01000, 1), (00100|00011|00001, 1), (11000|00111|00011, 1), (00100|11001|01001, 1), (10000|01001|
00001, 1), (01000|00111|00010, 1), (10000|01111|00001, 0.5), (10000|01111|00111, 0.5), (00010|00101|00001, 1), (00001|01110|00100, 1), (00010|11101|00100, 1)], [(01111, 0.5), (00111, 0.25), (00010,
0.25), (11000|00111|00011, 1), (00111|11000|01000, 1), (00100|00011|00001, 1), (00001|01110|00100, 1), (00010|00101|00001, 1), (00010|11101|00100, 1), (00100|11001|01001, 1), (01000|00111|00010, 1),
 (10000|01111|00001, 0.5), (10000|01111|00111, 0.5), (00100|01010|00010, 1), (10000|01001|00001, 1)] )
```

```
_build/gp_doctest.cpp:209: ERROR: CHECK_EQ( true_mean, inst.GetEngine()->GetBranchLengths()(2) ) is NOT correct!
  values: CHECK_EQ( 0.1268252727, 0.191715 )
```

### PR DESCRIPTION
**Pull request by 4ment**
## Description

Fixed tests failing on Mac in rooted_sbn_instance.hpp

Closes #280

### CODE REVIEW

File: src/rooted_sbn_instance.hpp [hunk 1]
```diff
@@ -361,10 +364,12 @@ TEST_CASE("RootedSBNInstance: reading SBN parameters from a CSV") {
   auto inst = MakeFiveTaxonRootedInstance();
   inst.ReadSBNParametersFromCSV("data/test_modifying_sbn_parameters.csv");
   auto pretty_indexer = inst.PrettyIndexer();
-  CHECK_EQ(pretty_indexer[10], "10000|01111|00001");
-  CHECK_LT(fabs(inst.sbn_parameters_[10] - log(0.15)), 1e-8);
+  auto it = find(pretty_indexer.begin(), pretty_indexer.end(), "10000|01111|00001");
```
Comment (matsen) [inline src/rooted_sbn_instance.hpp:367]: I'd prefer something like `gpcsp_it`.

File: src/rooted_sbn_instance.hpp [hunk 2]
```diff
@@ -361,10 +364,12 @@ TEST_CASE("RootedSBNInstance: reading SBN parameters from a CSV") {
...
+  CHECK(it != pretty_indexer.end());
+  auto index = std::distance(pretty_indexer.begin(), it);
```
Comment (matsen) [inline src/rooted_sbn_instance.hpp:369]: I'd prefer something like `gpcsp_idx`.

File: src/rooted_sbn_instance.hpp (hunk 1 above)
Comment (matsen) [reply src/rooted_sbn_instance.hpp:367]: Also, how does this compile just with `find` vs `std::find`?

File: src/rooted_sbn_instance.hpp [hunk 3]
```diff
@@ -185,7 +185,10 @@ TEST_CASE("RootedSBNInstance: subsplit support and TrainSimpleAverage") {
                                          {"00010|00101|00001", 1},
                                          {"00001|01110|00100", 1},
                                          {"00010|11101|00100", 1}});
-  CHECK_EQ(correct_parameters, inst.PrettyIndexedSBNParameters());
+  sort(correct_parameters.begin(), correct_parameters.end());
```
Comment (matsen) [inline src/rooted_sbn_instance.hpp:188]: How does this compile with `sort` vs `std::sort`? Did we open `std` somewhere?

**Review by matsen — COMMENTED** [review]
🙇 , @4ment ! This runs on my side. I'm happy to make my pedantic modifications if you don't want to-- just let me know.

File: src/rooted_sbn_instance.hpp (hunk 3 above)
Comment (4ment) [reply src/rooted_sbn_instance.hpp]: It is using argument dependent lookup to find `sort` in `std`. I changed `sort` to `std::sort` to make it explicit.

**Review by matsen — APPROVED** [review]

**Pull request merged**